storm-client/src/jvm/org/apache/storm/task/ShellBolt.java

```diff
@@ -351,6 +328,7 @@ public void run() {
 
     private class BoltReaderRunnable implements Runnable {
         public void run() {
+            _logHandler.setUpContext(_process, _context);
```

Comment:
This is effectively one-time setup, but if we really want to have one-time setup, maybe better to initialize it in prepare(), since we are more familiar with initializing everything in it.